common/src/test/org/apache/hive/common/util/TestDateParser.java

```diff
@@ -71,9 +71,6 @@ public void testParseDateFromTimestampWithCommonTimeDelimiter() {
   @Test
   public void testParseDateFromValidDateLiteralWithTrailingDigits() {
     assertEquals(Date.of(2023, 8, 8), DateParser.parseDate("2023-08-0800"));
-    // The result may seem unexpected but for many "08-08-20" is a valid date so there is no reason to reject
-    // "08-08-2023" and return null unless in the future Hive becomes stricter in terms of parsing dates.
-    assertEquals(Date.of(8, 8, 20), DateParser.parseDate("08-08-2023"));
```

Comment:
I prefer the new behavior, but it looks intentional. I have no objections to change it

Reply:
I guess we should ping @zabetak as it is him who implemented the test case
https://github.com/apache/hive/pull/4545